### ISSUE TITLE
Split-up ACL user/password into user/password and user/lockaccount

### ISSUE THREAD

**Issue by sunweaver** [gosa-project/gosa-core#2]
Hi,

This is a feature request.

currently, the permission to change other users' passwords and the permission to lock/unlock user accounts is handled by the same ACL (user/password).

It would be great to have two different ACLs here: e.g. user/password and user/lockaccount (or such).

Because: there are people that are eligit to change user password, but should rather not have the permission to fully lock down user accounts.

Greets,
Mike

**Comment by cajus** [comment]
Thats because locking/unlocking is done by modifying the password attribute. I'm not sure if these ACLs can be seperated at this point. Needs inspection.

**Comment by sunweaver** [comment]
Yeah, I noticed these non-standard password modififcations when locking users via GOsa...

OT here, but maybe worth investigating later...
For Samba, the "X" flag in sambaAcctFlags should be sufficient (no need to insert a "!" at the beginning of the password hashes).

For shadowAccounts, account locking should happen via shadow account flags.

For other LDAP services, the ppolicy overlay should be used in slapd together with the pwdAccountLockedTime attribute description (see http://www.openldap.org/lists/openldap-technical/200810/msg00107.html
).

For Kerberos, we (as in Debian Edu) currently lock the account via a hook script that calls kadmin.local (see:
http://anonscm.debian.org/cgit/debian-edu/debian-edu-config.git/commit/?id=d2bbb3979eb434efa095af8ff0ae9cfb2d10ce4f
).

**Comment by cajus** [comment]
GOsa locks passwords this way, because there was no effective other way to do so at the time of writing. There are already entries in the original bugtracker to migrate to pwPolicy, etc. The question is how existing services have evolved.

Also OT here, but we should create a docker (or vagrant or ...) image to have a reference implementation for testing all these techniques.
